interfaces/prompting/patterns/render_internal_test.go

```diff
@@ -95,6 +95,28 @@ func (s *renderSuite) TestRenderAllVariants(c *C) {
 	c.Check(variants, HasLen, parsed.NumVariants())
 }
 
+func (s *renderSuite) TestRenderAllVariantsConflicts(c *C) {
+	// all variants are the exact same path
+	pattern := "/{fo{obar},foo{b{ar,ar},bar,{ba}r}}"
+	scanned, err := scan(pattern)
+	c.Assert(err, IsNil)
+	parsed, err := parse(scanned)
+	c.Assert(err, IsNil)
+
+	expectedVariants := []string{
+		"/foobar",
+	}
+
+	variants := make([]string, 0, len(expectedVariants))
+	renderAllVariants(parsed, func(index int, variant PatternVariant) {
+		variants = append(variants, variant.String())
+	})
+
+	c.Check(variants, DeepEquals, expectedVariants)
+	c.Check(variants, HasLen, parsed.NumVariants())
+	c.Check(variants, HasLen, 1)
+}
```

Comment:
```suggestion
	// all variants are the exact same path
	for _, testCase := range []struct {
		pattern          string
		expectedVariants []string
	}{
		{
			"/{fo{obar},foo{b{ar,ar},bar,{ba}r}}",
			{"foobar"},
		},
		{
			"/{{foo/{bar,baz},123}{123,foo{/bar,/baz}}}",
			{
				"/foo/bar",
				"/foo/baz",
				"/123",
			},
		},
	} {
		scanned, err := scan(testCase.pattern)
		c.Assert(err, IsNil)
		parsed, err := parse(scanned)
		c.Assert(err, IsNil)

		variants := make([]string, 0, len(testCase.expectedVariants))
		renderAllVariants(parsed, func(index int, variant PatternVariant) {
			variants = append(variants, variant.String())
		})

		c.Check(variants, DeepEquals, testCase.expectedVariants)
		c.Check(variants, HasLen, parsed.NumVariants())
	}
}
```

Reply:
Might need to sort the expected variants and received variants as well, unless we figure out what the ordering ought to be.

Reply:
So I've tested this locally, the new suggestion is what actually happens, we end up with duplicate patterns. I will look into whether this is okay, but if it's not, we'll want to add a uniqueness check so we don't self-conflict by accident.